lib/parsers/index.ts

```diff
@@ -96,7 +97,8 @@ export async function getDependencyTreeFromPackagesConfig(manifestFile, includeD
     const depName = dep.$.id;
     const isDev = !!dep.$.developmentDependency;
     depTree.hasDevDependencies = depTree.hasDevDependencies || isDev;
-    if (isDev && !includeDev) {
+    if ((isDev && !includeDev) ||
+      (targetFramework && dep.$.targetFramework && dep.$.targetFramework !== targetFramework)) {
```

Comment:
It's not exactly true Ariel showed me an example when target framework in csproj file is something like `Profile111` and in project.config file is something like `portable-net45+win8+wpa81` which is the same.

https://blog.stephencleary.com/2012/05/framework-profiles-in-net.html

I'm not sure about it. But anyway it's a good idea to ensure if we can strictly compare name.

Reply:
@Kirill89 I believe this is probably the best we can do so far... I'm also not entirely sure, what's current status of our cross file (I mean getting target framework from csproj and passing it along with packages.config) support is.

Reply:
https://github.com/snyk/nuget-frameworks for that purpose. But I think we cannot use it here.

Reply:
@Kirill89 Btw: don't you think responsibility of translation between different naming lays outside of the changed function? I can imagine we will introduce, but then there's the responsibility of the caller to know, what should be expected in `packages.config`.

Reply:
Maybe you’re right. But comparator itself (‘!==‘) is inside the  parser.
Probably it’s better idea to return target framework as is (for each dependency) and filter root dependencies outside.

Reply:
@Kirill89 I thought about this option too, but I don't like - because we would need to have `packagesConfigDepTree` interface over `depTree` for this specific case, which doesn't sound right to me.

Reply:
But we have no other option. If we don't want to put here complex comparison logic we must make this available from outside.

Reply:
Annotate all deps and filtering should be done not in this lib at all but in dotn-net-deps?